datanode/dehistory/store/index.go

```diff
@@ -8,97 +8,125 @@ import (
 	"path/filepath"
 	"sort"
 
-	"github.com/syndtr/goleveldb/leveldb"
-	"github.com/syndtr/goleveldb/leveldb/opt"
-	"github.com/syndtr/goleveldb/leveldb/util"
+	bolt "go.etcd.io/bbolt"
 )
 
 var ErrIndexEntryNotFound = errors.New("index entry not found")
 
-type LevelDbBackedIndex struct {
-	db *leveldb.DB
+type BBoltBackedIndex struct {
+	db *bolt.DB
```

Comment:
Why the change away from levelDB? `resource temporarily unavailable` usually just means that `db.Close()` wasn't called properly when shutting down, then the next process trying to open it gets that error.

Also I'm sure on a previous PR it was decided to use levelDB to keep the tools consistent across all of core?

Reply:
ok, thats interesting, digging into the code it was not possible to get a handle on the cause as the error handling in code is terrible and adds no context to be able to diagnose the underlying cause - doesn't inspire confidence in the lib.

If we think that's the cause I'll pull the PR see if can figure out why close not being called.

Reply:
Where I've seen that error before was when doing the core-snapshot work -- there was a system-test which would stop the node, then call `vegatools snapshotdb` to look how many snapshots were saved. The call to `vegatools snapshotdb` would sometime fail with that resource unavailable error, and making sure we closed the db both in core on shutdown and in the tool fixed it. So it sounds very similar to this!